pkgs/top-level/all-packages.nix

```
@@ -6345,9 +6345,7 @@ with pkgs;
 
   agebox = callPackage ../tools/security/agebox { };
 
-  age-plugin-yubikey = callPackage ../tools/security/age-plugin-yubikey {
-    inherit (pkgs.darwin.apple_sdk.frameworks) Foundation PCSC;
-  };
+  age-plugin-yubikey = callPackage ../tools/security/age-plugin-yubikey { };
```

```suggestion
  age-plugin-yubikey = darwin.apple_sdk_11_0.callPackage ../tools/security/age-plugin-yubikey { };
```

You upgraded the darwin sdk version in this PR. If sdk_11 is actually a requirement now, then it is recommended to make this change. On Linux the effect will be the same as `callPackage`.

https://nixos.org/manual/nixpkgs/stable/#sec-darwin

---

The update is needed. Why is it recommended to do it that why? i really prefer the more explicitl way in this PR.

---

This doesn't replace your changes inside the package file, it replaces the calling function. As the manual says:

> x86_64-darwin uses the 10.12 SDK by default, but some software is not compatible with that version of the SDK. In that case, the 11.0 SDK used by aarch64-darwin is available for use on x86_64-darwin.

---

Does it not expose the frameworks at the top? Either way, I legit doesn't save me anything. I'd rather sdk version be specified inside the actual expression than outside of it.

---

Huh? This function is used 79 times in `all-packages.nix` alone, including for quite well maintained packages such as go itself.

My understanding of this function is that it allows users on x86_64-darwin to compile the package, by explicitly including the necessary sdk_11. Without it, those users will be unable to build this package since it now depends on that version of the sdk.

My build test for x86_64-darwin was run on aarch64-darwin.

---

GitHub search failed me once again. I still don't think it's needed considering the fact ofborg managed to build it fine on `x86_64-darwin`.

---

What do you gain from using `darwin.apple_sdk_11_0.callPackage` when you still have to refer to the frameworks with `apple_sdk_11_0` inside the expression? It seems to be building without it, no?